### PR TITLE
chore: New Builder Auth

### PR DESCRIPTION
**Pull request by istarkov**
### Full track

https://github.com/webstudio-is/webstudio-saas/issues/317#issuecomment-2289241819

So thoughts about session
https://github.com/webstudio-is/webstudio-saas/issues/317#issuecomment-2314515772

Some explanations what is fixed and how
https://github.com/webstudio-is/webstudio-saas/issues/317#issue-2465048823

Next PR:
- [x] - Old share links auto redirect
- [x] - Try make better error message having that current PR will logout all users.

## Steps for reproduction

1. Login using `google login` or `dev login` try to open projects on dahsboard, create new, clone delete etc.
2. Try to open projects on dahboard, see each is on it's own domain and work
3. Try to open project link (without share token) with other user - should fail.
4. Try publish/clone any links and operations.
5. Share project from `userA`:
 - try to open it with other `userB`
   - see `userB` is not logged in to that project
   - try to get using fetch from console anything about `userB` or `userA` i.e rest etc endpoints on apps.webstudio.is domain 
 - try to open incognito, check it works.
6. Open few project, then logout from dashboard, reload project page
 - See login page
 - Try to login as same uer - see it works
 - Or try to login as another user - see auth error

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env` file

### CODE REVIEW

**Comment by istarkov** [comment]
Tests:

1. Login via google
2. Create new Project, created, Im redirected to a new project.
3. Clone project using project menu. Cloned/redirected.
4. Edit text - working
5. Create new page and links - preview working.
6. Publish - working

File: apps/builder/app/builder/features/workspace/canvas-iframe.tsx [hunk 1]
```diff
@@ -23,7 +23,17 @@ export const CanvasIframe = forwardRef<HTMLIFrameElement, CanvasIframeProps>(
...
-    return <iframe {...props} ref={ref} className={iframeStyle()} />;
+    const credentialless = {
+      credentialless: "true",
+    };
+    return (
+      <iframe
+        {...props}
+        ref={ref}
+        className={iframeStyle()}
+        {...credentialless}
```
Comment (TrySound) [inline apps/builder/app/builder/features/workspace/canvas-iframe.tsx:34]: why not just prop?

Comment (istarkov) [reply apps/builder/app/builder/features/workspace/canvas-iframe.tsx:34]: typescript error ;)

Comment (istarkov) [reply apps/builder/app/builder/features/workspace/canvas-iframe.tsx:34]: decided to not extend interface

Comment (kof) [reply apps/builder/app/builder/features/workspace/canvas-iframe.tsx:34]: would be more obvious to silence ts in that place or at least add a comment ...

File: apps/builder/app/routes/leak.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { json, type LoaderFunction } from "@remix-run/server-runtime";
```
Comment (TrySound) [inline apps/builder/app/routes/leak.ts:1]: could you clarify what this route for?

Comment (istarkov) [reply apps/builder/app/routes/leak.ts:1]: leak testing, will remove

Comment (istarkov) [reply apps/builder/app/routes/leak.ts:1]: The idea that if you will be able to get information of the visitor using this links it will be a security bug

Comment (kof) [reply apps/builder/app/routes/leak.ts:1]: needs comment with repro steps

Comment (istarkov) [reply apps/builder/app/routes/leak.ts:1]: For now no repro, we need to try to find

Comment (istarkov) [reply apps/builder/app/routes/leak.ts:1]: It's like endpoint without any permissions check and even without it should not be possible (except copy-paste bad actor code to somehow see other user data)

Comment (istarkov) [reply apps/builder/app/routes/leak.ts:1]: see `message: "If you see other user then data leaked",`

**Review by TrySound — APPROVED** [review]

**Pull request merged**